packages/components/MenuButton/src/MenuButton.types.ts

```diff
@@ -13,8 +13,10 @@ export interface MenuButtonState {
 }
 
 export interface MenuButtonItemProps extends ContextualMenuItemProps {
-  submenu?: boolean;
+  hasSubmenu?: boolean;
```

Comment:
Feels like we could infer this from the existence of submenuItems/submenu? Though I guess it's better to be explicit.